cfripper/rules/stack_name_matches_regex.py

```diff
@@ -0,0 +1,40 @@
+from typing import Dict, Optional
+
+from pycfmodel.model.cf_model import CFModel
+
+from cfripper.config.regex import REGEX_ALPHANUMERICAL_OR_HYPHEN
+from cfripper.model.enums import RuleMode, RuleRisk
+from cfripper.model.result import Result
+from cfripper.rules.base_rules import Rule
+
+
+class StackNameMatchesRegexRule(Rule):
+    """
+    Checks that a given stack follows the naming convention given by a regex.
+    """
+
+    RULE_MODE = RuleMode.DEBUG
+    RISK_VALUE = RuleRisk.LOW
+    REASON = (
+        "The stack name {} does not follow the naming convention (only alphanumerical characters and hyphens allowed)."
+    )
+
+    def _stack_name_matches_regex(self, stack_name: str) -> bool:
+        """Check that stack name follows naming convention."""
+        return bool(REGEX_ALPHANUMERICAL_OR_HYPHEN.match(stack_name))
+
+    def invoke(self, cfmodel: CFModel, extras: Optional[Dict] = None) -> Result:
+        result = Result()
+        stack_name = self._config.stack_name
```

Comment:
The `stack_name` should either be in the config or in the extras.
Besides that, we need to add so in the description of the rule, even though the rule is only for demo purposes

Reply:
Not sure I understand the first part, we already get `stack_name` from the config?